[Controls] remove allow-listed Converters

StephaneDelcroix opened the pull request:
### Description of Change

- Move EasingTypeConverter to Core, attribute the type, and register the
compiled converter
- Remove Easing and Color converters from the allow-list

see https://github.com/dotnet/maui/issues/5660

eerhardt commented:
FYI - this will need to be merged with my recent change https://github.com/dotnet/maui/pull/5640.  Shouldn't be too bad though.

eerhardt commented on src/Controls/src/Core/BindableProperty.cs:
```diff
@@ -38,13 +38,6 @@ public sealed class BindableProperty
 
-		static readonly Dictionary<Type, TypeConverter> KnownTypeConverters = new Dictionary<Type, TypeConverter>
-		{
-			{ typeof(Uri), new UriTypeConverter() },
-			{ typeof(Color), new ColorTypeConverter() },
-			{ typeof(Easing), new EasingTypeConverter() },
```
The issue with `Uri` and `Easing` is that the converter isn't marked on the Type.

They are both defined in a library that can't reference the library the converter is in.

We could fix this in `Easing` using a string reference instead. But that isn't very safe nor performant since you need to use `Type.GetType` and hope the Type is around. Maybe we could move `EasingTypeConverter` from `Microsoft.Maui.Controls` to `Microsoft.Maui` to solve this?

But Uri will have to be solved a different way. Going through System.ComponentModel.TypeDescriptor.GetConverter, it does have this mapping here:

https://github.com/dotnet/runtime/blob/4ed3063d8bc434fd29fe63cf093c17baa9ff8fd3/src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/ReflectTypeDescriptionProvider.cs#L180

StephaneDelcroix replied:
you're right about UriTypeConverter, I've been too quick to clean stuffs. what worries me is that we didn't got any failing test :(

StephaneDelcroix replied:
EasingTypeConverter can be moved, easily, I'm doing it. We definitely should replace our build-in conversion logic for native types with TypeDescriptor, let's track that on #5660

eerhardt replied:
> what worries me is that we didn't got any failing test 

💯

StephaneDelcroix committed:
[Core|Controls]

- Move EasingTypeConverter to Core, attribute the type, and register the
compiled converter
- Remove Easing and Color converters from the whitelist

see #5660

rmarinho commented on src/Core/src/Converters/EasingTypeConverter.cs:
```diff
@@ -5,21 +5,18 @@
-	[Xaml.ProvideCompiled("Microsoft.Maui.Controls.XamlC.EasingTypeConverter")]
 	public class EasingTypeConverter : TypeConverter
 	{
-		/// <include file="../../docs/Microsoft.Maui.Controls/EasingTypeConverter.xml" path="//Member[@MemberName='CanConvertFrom']/Docs" />
```
DO we want to remove these or should we just update this to the right place @hartez  ?

rmarinho commented on src/Controls/src/Core/BindableProperty.cs:
```diff
@@ -41,8 +41,6 @@ public sealed class BindableProperty
 		static readonly Dictionary<Type, TypeConverter> KnownTypeConverters = new Dictionary<Type, TypeConverter>
 		{
 			{ typeof(Uri), new UriTypeConverter() },
-			{ typeof(Color), new ColorTypeConverter() },
```
Do we want to remove this one too ? or just the easing ?

StephaneDelcroix replied:
we want to remove this one

eerhardt reviewed (approved):
LGTM
